src/insane/mcmc_cfbd.jl

```diff
@@ -14,69 +14,777 @@ Created 07 10 2021
+                μi       ::Float64               = NaN,
+                ψi       ::Float64               = NaN,
+                pupdp    ::NTuple{4,Float64}     = (0.2,0.2,0.2,0.2),
+                prints   ::Int64                 = 5,
+                tρ       ::Dict{String, Float64} = Dict("" => 1.0))
+
+Run insane for constant fossilized birth-death.
+"""
+function insane_cfbd(tree     ::sTf_label, 
+                     out_file ::String;
+                     λ_prior  ::NTuple{2,Float64}     = (1.0, 1.0),
+                     μ_prior  ::NTuple{2,Float64}     = (1.0, 1.0),
+                     λmμ_prior::NTuple{2,Float64}     = (NaN, NaN),
+                     ψ_prior  ::NTuple{2,Float64}     = (1.0, 1.0),
+                     niter    ::Int64                 = 1_000,
+                     nthin    ::Int64                 = 10,
+                     nburn    ::Int64                 = 200,
+                     marginal ::Bool                  = false,
+                     nitpp    ::Int64                 = 100, 
+                     nthpp    ::Int64                 = 10,
+                     K        ::Int64                 = 11,
+                     ϵi       ::Float64               = 0.4,
+                     λi       ::Float64               = NaN,
+                     μi       ::Float64               = NaN,
+                     ψi       ::Float64               = NaN,
+                     pupdp    ::NTuple{4,Float64}     = (0.2,0.2,0.2,0.2),
+                     prints   ::Int64                 = 5,
+                     tρ       ::Dict{String, Float64} = Dict("" => 1.0))
+
+  n  = ntips(tree)
+  th   = treeheight(tree)
+  stem = !iszero(e(tree))
+
+  # set tips sampling fraction
+  if isone(length(tρ))
+    tl = tiplabels(tree)
+    tρu = tρ[""]
+    tρ = Dict(tl[i] => tρu for i in 1:n)
+  end
+
+  # make fix tree directory
+  idf = make_idf(tree, tρ)
+
+  # starting parameters
+  if isnan(λi) && isnan(μi) && isnan(ψi)
+    # if only one tip
+    if isone(n)
+      λc = prod(λ_prior)
+      μc = isnan(μ_prior[1]) ? prod(λ_prior)-prod(λmμ_prior) : prod(μ_prior)
+    else
+      λc, μc = moments(Float64(n), ti(idf[1]), ϵi)
+    end
+    # if no sampled fossil
+    if iszero(nfossils(tree))
+      ψc = prod(ψ_prior)
+    else
+      ψc = nfossils(tree)/treelength(tree)
+    end
   else
-    @error string(augmentation," does not match `fs` or `gp`")
+    λc, μc, ψc = λi, μi, ψi
+  end
+  # M attempts of survival
+  mc = m_surv_cbd(th, λc, μc, 1_000, stem)
+
+  # make a decoupled tree and fix it
+  Ξ = sTfbd[]
+  sTfbd!(Ξ, tree)
+
+  # make parameter updates scaling function for tuning
+  spup = sum(pupdp)
+  pup  = Int64[]
+  for i in Base.OneTo(4)
+    append!(pup, fill(i, ceil(Int64, Float64(2*n - 1) * pupdp[i]/spup)))
   end
 
-  return R, tree
+  @info "Running constant fossilized birth-death with forward simulation"
+
+  # adaptive phase
+  @info "MCMC - Adaptive phase"
```

Comment:
If you use progress bar this info would be redundant. Let's keep the information messages congruent across models ;)